packages/playground/src/data/playground-api-client.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  ApiResponse,
+  CreateAccountRequest,
+  CreateAccountResponseData,
+  ErrorResponse
+} from "@counterfactual/playground-server";
+
+const BASE_URL = `ENV:API_HOST`;
+
+type JsonBody = {
+  [key: string]: string | number | boolean | JsonBody | undefined;
+};
```

Comment:
Seems like a super common type. Is this not in some generic typings package or Typescript built-in somewhere?

Reply:
IIRC `undefined` is not a valid value in JSON, the equivalent is `null`.

Reply:
@snario It's a type generic enough to avoid doing.

```ts
type ApiResponse = CreateAccountResponseData | LoginResponseData | ...ResponseData;
```

The standard is to use `any`, `object` or `{ [key: string]: any }` - but they seem too permissive to me.

Reply:
@cf19drofxots Actually `null` isn't equivalent to `undefined`. `undefined` means a key isn't defined at all in the object, while `null` means a key is present and has a null value.

Reply:
In this case, it's used because a key attempting to be accessed maybe undefined.

Reply:
@joelalejandro you're right. I was just pointing out that `undefined` isn't supported in JSON and we seem to be using it as part of the JsonBody definition.

Reply:
@cf19drofxots Yeah, that's true, but it's valid in terms of a proper result for an unaccessible member.